.evergreen/install-dependencies.sh

```diff
@@ -1,108 +1,98 @@
+if [[ $architecture = "x86_64" ]]; then
+  architecture="x64"
+elif [[ $architecture = "arm64" ]]; then
+  architecture="arm64"
+elif [[ $architecture = "aarch64" ]]; then
+  architecture="arm64"
+elif [[ $architecture == s390* ]]; then
+  architecture="s390x"
+elif [[ $architecture == ppc* ]]; then
+  architecture="ppc64le"
+else
+  echo "Unable to determine operating system: $architecture"
+  exit 1
+fi
 
+file_extension="tar.gz"
+if [[ "$OS" = "Windows_NT" ]]; then file_extension="zip"; fi
 
-  NVM_HOME=$(cygpath -w "$NVM_DIR")
-  export NVM_HOME
-  NVM_SYMLINK=$(cygpath -w "$NODE_ARTIFACTS_PATH/bin")
-  export NVM_SYMLINK
-  NVM_ARTIFACTS_PATH=$(cygpath -w "$NODE_ARTIFACTS_PATH/bin")
-  export NVM_ARTIFACTS_PATH
-  PATH=$(cygpath $NVM_SYMLINK):$(cygpath $NVM_HOME):$PATH
-  export PATH
+node_directory="node-${node_index_version}-${operating_system}-${architecture}"
+node_archive="${node_directory}.${file_extension}"
+node_archive_path="$NODE_ARTIFACTS_PATH/${node_archive}"
+node_download_url="https://nodejs.org/dist/${node_index_version}/${node_archive}"
 
-  curl -L $NVM_WINDOWS_URL -o nvm.zip
-  unzip -d "$NVM_DIR" nvm.zip
-  rm nvm.zip
+echo "Node.js ${node_index_version} for ${operating_system}-${architecture} released on ${node_index_date}"
 
-  chmod 777 "$NVM_DIR"
-  chmod -R a+rx "$NVM_DIR"
+set -o xtrace
 
-  cat <<EOT > "$NVM_DIR/settings.txt"
-root: $NVM_HOME
-path: $NVM_SYMLINK
-EOT
-  nvm install "$NODE_VERSION"
-  nvm use "$NODE_VERSION"
-  which node || echo "node not found, PATH=$PATH"
-  which npm || echo "npm not found, PATH=$PATH"
-  npm cache clear --force # Fixes: Cannot read properties of null (reading 'pickAlgorithm') error on windows
+curl --fail --retry 8 -sS "${node_download_url}" --max-time 300 --output "$node_archive_path"
 
-# install Node.js on Linux/MacOS
+if [[ "$file_extension" = "zip" ]]; then
+  unzip -q "$node_archive_path" -d "${NODE_ARTIFACTS_PATH}"
+  mkdir -p "${NODE_ARTIFACTS_PATH}/nodejs"
+  # Windows "bins" are at the top level
+  mv "${NODE_ARTIFACTS_PATH}/${node_directory}" "${NODE_ARTIFACTS_PATH}/nodejs/bin"
+  # Need to add executable flag ourselves
+  chmod +x "${NODE_ARTIFACTS_PATH}/nodejs/bin/node.exe"
+  chmod +x "${NODE_ARTIFACTS_PATH}/nodejs/bin/npm"
 else
-  curl -o- $NVM_URL | bash
-  [ -s "${NVM_DIR}/nvm.sh" ] && source "${NVM_DIR}/nvm.sh"
-  nvm install --no-progress "$NODE_VERSION"
+  tar -xf "$node_archive_path" -C "${NODE_ARTIFACTS_PATH}"
+  mv "${NODE_ARTIFACTS_PATH}/${node_directory}" "${NODE_ARTIFACTS_PATH}/nodejs"
 fi
 
-npm install ${NPM_OPTIONS}
+export PATH="$NODE_ARTIFACTS_PATH/npm_global/bin:$NODE_ARTIFACTS_PATH/nodejs/bin:$PATH"
+hash -r
+
+# Set npm -g prefix to our local artifacts directory
+cat <<EOT > .npmrc
+prefix=$NODE_ARTIFACTS_PATH/npm_global
+EOT
+
+# Cannot upgrade npm version for node 12
+if [[ $operating_system != "win" ]] && [[ $NODE_LTS_NAME != "erbium" ]]; then
+  # Update npm to latest when we can
+  npm install --global npm@latest
+  hash -r
+fi
```

Comment:
Had to exempt 12 from upgrading npm, should we try to upgrade to npm v8 instead?

Reply:
Seems like that would be a good idea. What's it on 12 by default? 6.x?

Reply:
Yep 12.9.1 ships with 6.10.2, but it should be able to run up to 8

I'll update this to install `@8` just for node 12 tasks

https://www.npmjs.com/package/npm?activeTab=versions